### PR TITLE
[#406] Add component: notifications

### LINKED ISSUE #406: Add component: notifications
Notification cards to appear top-right, to inform the user of an action that has happened (e.g. “Password updated successfully”), with an icon on the left.

Variants:
- dismiss button top right
- ok/cancel buttons on the bottom
- various icon colors

### PR DESCRIPTION
**Pull request by planktonic**
Fixes #406 

Adds a simple notification component, and a notification with buttons.

Looks like:

Screenshot 2021-03-10 at 16 34 11

Screenshot 2021-03-10 at 16 34 20

### CODE REVIEW

File: scss/bitstyles/atoms/button--icon-reversed/_.scss [hunk 1]
```diff
@@ -14,23 +14,23 @@
...
   }
 
   &:disabled {
     color: $bitstyles-button-icon-reversed-color-disabled;
     cursor: default;
     background: $bitstyles-button-icon-reversed-background-color-disabled;
-    border: $bitstyles-button-icon-reversed-border-disabled;
+    border-color: $bitstyles-button-icon-reversed-border-color-disabled;
     box-shadow: $bitstyles-button-icon-reversed-box-shadow-disabled;
   }
```
Comment (andreasknoepfle) [inline scss/bitstyles/atoms/button--icon-reversed/_.scss:35]: So does that mean that we are now adding a border to all the buttons--icons and button--dangers that was not there before?

**Review by andreasknoepfle — APPROVED** [review]
The notification itself looks great :) 
This is meant to be displayed in the top-right corner, right? 

While this is great for those SPA-style apps, to display events, there are still some use-cases for classic web pages, where we might display the notification more present, since it is the result of an immediate action after a page load. 

Examples: 

- Login fails --> We do not want to show exactly what failed, for security reasons 
- Buttons without form that can still fail (certain actions make it impossible to hide the button if it would result in an error, beforehand). I know one wants to avoid this, but in real life admin backends this is normal (network issues, conflicts, other....)

File: scss/bitstyles/atoms/button--icon-reversed/_.scss (hunk 1 above)
Comment (planktonic) [reply scss/bitstyles/atoms/button--icon-reversed/_.scss]: Aha, no that’s from #411 — that got rebased but I didn’t update this branch. I’ve rebased! That PR will need to be merged before this one

(The change here is that border-color can be changed on hover etc, but all buttons will have the same _width_ of border. This means sizing and vertical alignment is equal, and no shifting happens on hover/etc)

**Comment by planktonic** [comment]
@andreasknoepfle yes this a top-right kinda notification. It’s meant for any events, but if the event is form-related or occurs after a page reload, a “flash”-style would be more noticeable 👍 

I’ll add a flash component in this branch, and a couple of examples:

[] - global flash, for “Login failed” etc
[] - flash in a form, to indicate something was invalid (field-specific feedback like “Email address invalid” should still render next to the relevant field though)

I’ll assume for now that there will be no “dismiss notification” button. Did I miss anything from that list?

**Comment by planktonic** [comment]
I’ve added a flash component, looks like:

Screenshot 2021-03-15 at 13 00 02

Screenshot 2021-03-15 at 13 00 08

**Review by andreasknoepfle — APPROVED** [review]
Absolute awesomeness 💯

**Pull request merged**